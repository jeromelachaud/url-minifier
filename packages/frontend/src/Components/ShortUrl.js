import React, { Component } from 'react'
import PropTypes from 'prop-types'
import './ShortUrl.css'

export default class ShortUrl extends Component {
  render() {
    const {
      link,
      message,
    } = this.props
    return (
      <div
        className="short-url">
        <div
          className="short-url__message">
          {message}
        </div>
        <a
          className="short-url__link"
          href={link}>
          {link}
        </a>
      </div>
    )
  }
}

ShortUrl.propTypes = {
  link: PropTypes.string.isRequired,
  message: PropTypes.string.isRequired,
}
