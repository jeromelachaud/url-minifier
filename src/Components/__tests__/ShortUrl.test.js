import React from 'react'
import { shallow } from 'enzyme'
import ShortUrl from '../ShortUrl'

const link = 'string'
const message = 'string'

it('renders correctly', () => {
  const tree = shallow(
    <ShortUrl
      link={link}
      message={message}
    />
  )
  expect(tree).toMatchSnapshot()
})
