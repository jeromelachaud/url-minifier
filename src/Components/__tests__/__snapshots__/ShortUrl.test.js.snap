// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="short-url"
>
  <div
    className="short-url__message"
  >
    string
  </div>
  <a
    className="short-url__link"
    href="string"
    target="_blank"
  >
    string
  </a>
</div>
`;
